src/jmcomic_ai/core.py

```
@@ -302,51 +307,229 @@ def get_ranking(self, period: str = "day", page: int = 1) -> dict[str, Any]:
+                        asyncio.run_coroutine_threadsafe(ctx.info(msg), loop)
+
+        # 3. Blocking Download Function
+        def _blocking_download():
             try:
-                self.logger.info(f"Starting download for album {album_id} to {target_path}")
-                self.option.download_album(album_id)
+                self.logger.info(f"Starting blocking download for album {album_id}")
+                # Pass the class, jmcomic will instantiate it with self.option
+                self.option.download_album(album_id, downloader=McpProgressDownloader)
                 self.logger.info(f"Download completed for album {album_id}")
+                return "success"
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)